autoprecompiles/src/memory_optimizer.rs

```diff
@@ -99,8 +105,9 @@ impl<T: FieldElement> MemoryBusInteraction<T> {
     /// Otherwise returns `Ok(Some(memory_bus_interaction))`
     fn try_from_symbolic_bus_interaction(
         bus_interaction: &SymbolicBusInteraction<T>,
+        bus_map: &BusMap,
     ) -> Result<Option<Self>, ()> {
-        if bus_interaction.id != MEMORY_BUS_ID {
+        if bus_interaction.id != bus_map.get_bus_id(&BusType::Memory).unwrap() {
```

Comment:
I think it would be better to supply the bus id as argument instead of the bus map